src/ChronosDate.php

```diff
@@ -115,8 +142,416 @@ public function __construct(
             $testNow = $testNow->modify($time);
         }
 
-        $time = $testNow->format('Y-m-d 00:00:00');
-        parent::__construct($time);
+        return new DateTimeImmutable($testNow->format('Y-m-d 00:00:00'));
+    }
+
+    /**
+     * Get the difference formatter instance or overwrite the current one.
+     *
+     * @param \Cake\Chronos\DifferenceFormatterInterface|null $formatter The formatter instance when setting.
+     * @return \Cake\Chronos\DifferenceFormatterInterface The formatter instance.
+     */
+    public static function diffFormatter(?DifferenceFormatterInterface $formatter = null): DifferenceFormatterInterface
+    {
+        if ($formatter === null) {
+            if (static::$diffFormatter === null) {
+                static::$diffFormatter = new DifferenceFormatter();
+            }
+
+            return static::$diffFormatter;
+        }
+
+        return static::$diffFormatter = $formatter;
+    }
+
+    /**
+     * Returns the difference between this instance and target.
+     *
+     * @param \Cake\Chronos\ChronosDate $target Target instance
+     * @param bool $absolute Whether the interval is forced to be positive
+     * @return \DateInterval
+     */
+    public function diff(ChronosDate $target, bool $absolute = false): DateInterval
+    {
+        return $this->native->diff($target->native, $absolute);
+    }
+
+    /**
+     * Returns formatted date string according to DateTimeImmutable::format().
+     *
+     * @param string $format String format
+     * @return string
+     */
+    public function format(string $format): string
+    {
+        return $this->native->format($format);
+    }
+
+    /**
+     * Returns the timezone offset.
+     *
+     * @return int
+     */
+    public function getOffset(): int
+    {
+        return $this->native->getOffset();
+    }
+
+    /**
+     * Gets the Unix timestamp for this instance.
+     *
+     * @return int
+     */
+    public function getTimestamp(): int
+    {
+        return $this->native->getTimestamp();
+    }
+
+    /**
+     * Return time zone set for this instance.
+     *
+     * @return \DateTimeZone|false
+     */
+    public function getTimezone(): DateTimeZone|false
+    {
+        return $this->native->getTimezone();
+    }
```

Comment:
I don't think dates should have Timezones. This was a mistake in the previous design that we could correct now. I'm happy to do that as a follow up pull request if you'd prefer.

Reply:
I was just re-creating the api from the interface initially, but we should drop this and see what needs to be fixe in cake core.